deepchecks/nlp/datasets/classification/tweet_emotion.py

```diff
@@ -0,0 +1,114 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Dataset containing tweets and meta data information for multiclass prediction for emotion of tweet.
+
+The data has 4653 tweet records with 5 features and a multiclass target column, referring to the emotion of the tweet.
+
+This dataset is an extension of Cardiff's tweet_eval dataset,
+For additional details about the dataset, please refer to the original source: https://github.com/cardiffnlp/tweeteval
+"""
+
+import typing as t
+
+import numpy as np
+import pandas as pd
+
+from deepchecks.nlp import TextData
+
+__all__ = ['load_data', 'load_embeddings', 'load_precalculated_predictions']
+
+_FULL_DATA_URL = 'https://ndownloader.figshare.com/files/39265559'
+_EMBEDDINGS_URL = 'https://ndownloader.figshare.com/files/39264332'
+_PREDICTIONS_URL = 'https://ndownloader.figshare.com/files/39264461'
+_target = 'label'
+
+
+def load_data(data_format: str = 'TextData', as_train_test: bool = True) -> \
```

Comment:
Save the data locally and then load it from local if already exists? That's what we usually do

Reply:
Agreed